provider.go

```diff
@@ -242,6 +245,43 @@ func (g *InstanceGroup) createInstance(ctx context.Context) (string, error) {
 		}
 	}
 
+	if g.VolumeSize != 0 && g.VolumeType != "" {
+		volumeOpts := volumes.CreateOpts{
+			Name: spec.Name,
+			Size: g.VolumeSize,
+			VolumeType: g.VolumeType,
+			ImageID: spec.ImageRef,
+		}
+		volume, err := g.client.CreateVolume(ctx, volumeOpts)
+		if err != nil {
+			return "", err
+		}
+
+		for {
```

Comment:
https://github.com/gophercloud/gophercloud/blob/main/openstack/blockstorage/v3/volumes/util.go#L10

You can use `blockstorage.WaitForStatus()` instead.